internal/database/sqlcommon/operation_sql.go

```diff
@@ -181,22 +181,29 @@ func (s *SQLCommon) UpdateOperation(ctx context.Context, ns string, id *fftypes.
 		sq.Eq{"namespace": ns},
 	})
 
-	_, err = s.updateTx(ctx, operationsTable, tx, query, func() {
+	ra, err := s.updateTx(ctx, operationsTable, tx, query, func() {
 		s.callbacks.UUIDCollectionNSEvent(database.CollectionOperations, core.ChangeEventTypeUpdated, ns, id)
 	})
 	if err != nil {
 		return err
 	}
+	if ra < 1 {
+		return i18n.NewError(ctx, coremsgs.Msg404NoResult)
+	}
 
 	return s.commitTx(ctx, tx, autoCommit)
 }
 
 func (s *SQLCommon) ResolveOperation(ctx context.Context, ns string, id *fftypes.UUID, status core.OpStatus, errorMsg string, output fftypes.JSONObject) (err error) {
-	update := database.OperationQueryFactory.NewUpdate(ctx).
-		Set("status", status).
-		Set("error", errorMsg)
+	update := database.OperationQueryFactory.NewUpdate(ctx).S()
+	if status != "" {
+		update = update.Set("status", status)
+	}
+	if errorMsg != "" {
+		update = update.Set("error", errorMsg)
```

Comment:
Looks like it is not possible to clear an error message once one is set. It's probably fine, but just pointing it out.

Reply:
This is a good point. Now splitting out to an update DTO, we can have a string pointer to make this distinction.